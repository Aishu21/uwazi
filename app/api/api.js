import bodyParser from 'body-parser'
import db_config from './config/database.js'

export default app => {

  //set db to use
  db_config.db_url = db_config.production;

  //common middlewares
  app.use(bodyParser.json());

  //module routes
  require('./auth/routes.js')(app);

}
