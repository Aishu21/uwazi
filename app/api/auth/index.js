import needsAuthorization from './authMiddleware';

export { needsAuthorization };
export { default as encryptPassowrd } from './encryptPassword';
export { comparePasswords } from './encryptPassword';
