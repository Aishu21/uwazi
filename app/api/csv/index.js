import csvLoader from './csvLoader.js';

export default csvLoader;
