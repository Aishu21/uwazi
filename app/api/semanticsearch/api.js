import fetch from 'isomorphic-fetch';
import SEMANTIC_SEARCH_URL from 'api/config/semanticsearch';

const semanticSearchAPI = {
  async processDocument(args) {
    const res = await fetch(`${SEMANTIC_SEARCH_URL}/semanticSearch/searchOneDoc`, {
      method: 'POST',
      body: JSON.stringify(args),
      headers: {
        'Content-Type': 'application/json'
      }
    });
    const body = await res.json();
    return body;
  }
};

export default semanticSearchAPI;
