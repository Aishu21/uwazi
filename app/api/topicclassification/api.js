/** @format */

import 'isomorphic-fetch';
import TOPIC_CLASSIFICATION_URL from 'api/config/topicclassification';

const topicClassificationAPI = {
  async processDocument({ seq, model }) {
    const url = new URL(TOPIC_CLASSIFICATION_URL);
    url.searchParams.set('model', model);
    const res = await fetch(TOPIC_CLASSIFICATION_URL, {
      method: 'POST',
      body: JSON.stringify(seq),
      headers: {
        'Content-Type': 'application/json',
      },
    });
    const body = await res.json();
    return body;
  },
};

export default topicClassificationAPI;
