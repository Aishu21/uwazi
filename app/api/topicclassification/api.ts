/** @format */

import 'isomorphic-fetch';
import isReachable from 'is-reachable';
import { tcServer } from 'api/config/topicClassification';
import { URL } from 'url';
import request from 'shared/JSONRequest';

export async function getModels() {
  const tcUrl = new URL('models', tcServer);
  const msTimeout = 1000;
  if (!(await isReachable(tcUrl.href, { timeout: msTimeout }))) {
    // TODO: move this backend check to server start-up time, maybe
    return {
      models: {},
      error: `Topic Classification server is unreachable (${msTimeout})`,
    };
  }
  return request
    .get(tcUrl.href)
    .then(async res => res.json)
    .catch(err => ({
      models: {},
      error: `Error from topic-classification server: ${err.toString()}`,
    }));
}

export async function checkModelReady(arg: { model: string }) {
  const tcUrl = new URL('models', tcServer);
  tcUrl.searchParams.set('model', arg.model);
  const msTimeout = 1000;
  if (!(await isReachable(tcUrl.href, { timeout: msTimeout }))) {
    // TODO: move this backend check to server start-up time, maybe
    return { models: {}, error: `Topic Classification server is unreachable (${msTimeout})` };
  }
  return request
    .get(tcUrl.href)
    .then(async res => res.json)
    .catch(err => ({
      models: '',
      error: `Error from topic-classification server: ${err.toString()}`,
    }));
}
