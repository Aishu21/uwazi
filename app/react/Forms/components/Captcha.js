import React, { Component } from 'react';
import PropTypes from 'prop-types';


class Captcha extends Component {
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
    const { refresh } = this.props;
    this.state = { captchaUrl: '/captcha' };
    refresh(this.refresh.bind(this));
  }

  onChange(e) {
    const { onChange } = this.props;
    onChange(e.target.value);
  }

  refresh() {
    this.setState({ captchaUrl: `/captcha?v=${Math.random() * 1000}` });
  }

  render() {
    const { value } = this.props;
    const { captchaUrl } = this.state;
    return (
      <div className="captcha">
        <img src={captchaUrl} alt="captcha"/>
        <input
          className="form-control"
          onChange={this.onChange}
          value={value}
        />
      </div>
    );
  }
}

Captcha.defaultProps = {
  value: '',
  refresh: () => {},
};

Captcha.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  refresh: PropTypes.func,
};

export default Captcha;
