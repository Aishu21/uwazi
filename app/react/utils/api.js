import { browserHistory } from 'react-router';

import { isClient } from 'app/utils';
import { notify } from 'app/Notifications/actions/notificationsActions';
import { store } from 'app/store';
import loadingBar from 'app/App/LoadingProgressBar';

import { APIURL } from '../config.js';
import request from '../../shared/JSONRequest';

let cookie;
let locale;
let API_URL = APIURL;

const doneLoading = (data) => {
  loadingBar.done();
  return data;
};

const handleError = (e, endpoint) => {
  const error = e;
  error.endpoint = endpoint;

  if (!isClient) {
    return Promise.reject(error);
  }

  doneLoading();

  if (error.status === 401) {
    browserHistory.replace('/login');
  }

  if (error.status === 404) {
    browserHistory.replace('/404');
  }

  if (error.status === 500) {
    store.dispatch(notify('An error has occurred', 'danger'));
  }

  if (![500, 404, 401].includes(error.status)) {
    store.dispatch(notify(error.json.error, 'danger'));
  }

  return Promise.reject(error);
};

const _request = (url, data, method) => {
  loadingBar.start();
  return request[method](API_URL + url, data, { 'Content-language': locale, Cookie: cookie, 'X-Requested-With': 'XMLHttpRequest' })
  .then(doneLoading)
  .catch(e => handleError(e, { url, method }));
};

export default {
  get: (url, data) => _request(url, data, 'get'),

  post: (url, data) => _request(url, data, 'post'),

  delete: (url, data) => _request(url, data, 'delete'),

  cookie(c) {
    cookie = c;
  },

  locale(key) {
    locale = key;
  },

  APIURL(url) {
    API_URL = url;
  }
};
