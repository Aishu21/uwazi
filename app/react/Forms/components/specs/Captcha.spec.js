import React from 'react';
import { shallow } from 'enzyme';
import api from '../../../utils/api';
import Captcha from '../Captcha';

describe('Captcha', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      onChange: () => {},
    };
  });

  const render = () => {
    component = shallow(<Captcha {...props} />);
  };

  it('should render a Captcha image with an input', () => {
    render();
    expect(component).toMatchSnapshot();
  });

  describe('refresh()', () => {
    it('should return the refresh captcha method', done => {
      let refreshCaptcha;
      props.refresh = _refreshCaptcha => {
        refreshCaptcha = _refreshCaptcha;
      };
      render();
      expect(component.find('div div').props().dangerouslySetInnerHTML).toEqual({ __html: '' });
      spyOn(api, 'get').and.returnValue(Promise.resolve({ captcha: 'captchasvg', id: 2 }));
      refreshCaptcha();
      component.update();
      expect(component.find('div div').props().src).not.toBe('/api/captcha');
      done();
    });
  });
});
