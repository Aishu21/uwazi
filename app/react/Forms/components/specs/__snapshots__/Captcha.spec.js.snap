// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Captcha should render a Captcha image with an input 1`] = `
<div
  className="captcha"
>
  <img
    alt="captcha"
    src="/captcha"
  />
  <input
    className="form-control"
    onChange={[Function]}
    value=""
  />
</div>
`;
